### PR TITLE
doc, gen_vimdoc.py

### PR DESCRIPTION
**Pull request by justinmk**
`scripts/gen_vimdoc.py` (renamed from `gen_vimdoc.py`) now supports nested elements and markdown-formatted lists. 

Note: Doxygen does a pretty nice job of interpreting the quasi-Markdown in our doctsrings, but it can't make sense when we do things like this:

```
- foo1
  - foo2
zub
- bar1
```

The guideline is: if you want indented things, they must be lists. That `zub` in the above tree is parsed by Doxygen as belonging to the `foo2` element. Adding line breaks causes different problems... Easiest solution is for the docstring author to make `zub` a list item (`- zub`) with `- bar1` is nested in it.

todo

- [x] https://github.com/neovim/neovim/pull/9710

### CODE REVIEW

File: src/nvim/api/buffer.c [hunk 1]
```diff
@@ -972,7 +972,7 @@ Integer nvim_buf_add_highlight(Buffer buffer,
 /// To clear the namespace in the entire buffer, pass in 0 and -1 to
 /// line_start and line_end respectively.
 ///
-/// @param buffer     Buffer handle
+/// @param buffer     Buffer handle, or 0 for current buffer
```
Comment (bfredl) [inline src/nvim/api/buffer.c]: Why change this on one function only?

Comment (justinmk) [reply src/nvim/api/buffer.c]: I'll change it for all cases. I was trying to use this one for a script and noticed it wasn't specified.

**Comment by mhinz** [comment]
https://github.com/neovim/neovim/pull/9722#pullrequestreview-213515458

**Comment by bfredl** [comment]
@mhinz no need to change that,`winhl=Normal:XXX` works perfectly well for any window. (No need to make floats seem needing special treatment when they don't )

File: runtime/doc/deprecated.txt [hunk 2]
```diff
@@ -63,6 +63,15 @@ Options ~
...
 *'viminfo'*		Deprecated alias to 'shada' option.
 
+UI extensions~
+*ui-wildmenu*		Use `ext_cmdline` and `ext_popupmenu` instead.
+			Enabled by `ext_wildmenu` |ui-options|.
+			If `ext_wildmenu` is set, these events are emitted for
+			backwards-compatibility:
+				["wildmenu_show", items]
+				["wildmenu_select", selected]
+				["wildmenu_hide"]
```
Comment (justinmk) [inline runtime/doc/deprecated.txt]: @bfredl I moved this to deprecated.txt.

File: src/nvim/api/vim.c [hunk 3]
```diff
@@ -1306,14 +1309,14 @@ Array nvim_get_api_info(uint64_t channel_id)
...
 ///     versions of nvim!
 ///
-/// @param attributes Informal attributes describing the client. Clients might
-///                   define their own keys, but the following are suggested:
-///     - "website" Website of client (for instance github repository)
-///     - "license" Informal description of the license, such as "Apache 2",
-///                 "GPLv3" or "MIT"
-///     - "logo"    URI or path to image, preferably small logo or icon.
-///                 .png or .svg format is preferred.
+/// @param attributes Arbitrary string:string map of informal client properties.
```
Comment (justinmk) [inline src/nvim/api/vim.c]: Restricting this to string-only values. https://github.com/neovim/go-client/pull/36#issuecomment-476418658

File: runtime/doc/map.txt [hunk 4]
```diff
@@ -274,27 +274,31 @@ Using 0x80 as a single byte before other text does not work, it will be seen
...
 Unlike <expr> mappings, there are no special restrictions on the <Cmd>
 command: it is executed as if an (unrestricted) |autocmd| was invoked or an
 async event event was processed.
 
-In select-mode, |:map| and |:vmap| command mappings are executed in
-visual-mode.  Use |:smap| to handle select-mode.
+Note:
+- Because <Cmd> avoids mode-changes (unlike ":") it does not trigger
+  |CmdlineEnter| and |CmdlineLeave| events. This helps performance.
+- For the same reason, |keycodes| like <C-R><C-W> are interpreted in the
```
Comment (bfredl) [inline runtime/doc/map.txt]: @justinmk I don't think this is correct, try

    noremap <f2> <cmd>echo "<c-r><c-w>"<cr>
    noremap! <f2> <cmd>echo "<c-r><c-w>"<cr>

and press `<f2>` in various modes. The result is always the same, `<c-r><c-w>` are always interpreted as the bytes they represent, not according to any mode.

Comment (justinmk) [reply runtime/doc/map.txt]: 127e13f53ee0

**Pull request merged**